compiler/rustc_target/src/asm/mod.rs

```diff
@@ -712,3 +712,183 @@ pub fn allocatable_registers(
         }
     }
 }
+
+#[derive(
+    Copy,
+    Clone,
+    Encodable,
+    Decodable,
+    Debug,
+    Eq,
+    PartialEq,
+    PartialOrd,
+    Hash,
+    HashStable_Generic
+)]
+pub enum InlineAsmClobberAbi {
+    X86,
+    X86_64Win,
+    X86_64SysV,
+    Arm,
+    AArch64,
+    RiscV,
+}
+
+impl InlineAsmClobberAbi {
+    /// Parses a clobber ABI for the given target, or returns a list of supported
+    /// clobber ABIs for the target.
+    pub fn parse(
+        arch: InlineAsmArch,
+        target: &Target,
+        name: Symbol,
+    ) -> Result<Self, &'static [&'static str]> {
```

Comment:
I think it is somewhat important that we utilize `Target::adjusted_abi` here so that e.g. ABIs such as `system` and `efiapi` work out of the box here.

Reply:
I tried using `Abi` directly but the code got a bit messy. In any case we only support a small subset of ABIs for `clobber_abi`, I'm not sure if it makes sense to use the existing ABI machinery.